catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostIngestPlugin.java

```diff
@@ -115,63 +118,104 @@
+            .map(Update::getNewMetacard)
+            .collect(Collectors.toList()),
+        updateResponse.getRequest().getProperties());
+    return updateResponse;
+  }
 
-    process(updateStorageRequest.getContentItems());
-    return updateStorageRequest;
+  @Override
+  public DeleteResponse process(DeleteResponse deleteResponse) throws PluginExecutionException {
+    return deleteResponse;
   }
 
-  private boolean isNitfMimeType(String rawMimeType) {
-    try {
-      return NITF_MIME_TYPE.match(rawMimeType);
-    } catch (MimeTypeParseException e) {
-      LOGGER.debug("unable to compare mime types: {} vs {}", NITF_MIME_TYPE, rawMimeType);
+  private void updateContent(List<Metacard> metacards, Map<String, Serializable> properties) {
+    for (Metacard mcard : metacards) {
+      if (shouldGenerateContentItems(mcard, properties)) {
+        try {
+          createContentItems(mcard);
+        } catch (Exception e) {
+          LOGGER.warn("Error creating thumbnail/overview/original for {}", mcard.getId());
+        }
+      }
     }
+  }
 
-    return false;
+  private boolean shouldGenerateContentItems(
+      Metacard metacard, Map<String, Serializable> properties) {
+    Attribute type = metacard.getAttribute(Media.TYPE);
+    return type != null
+        && IMAGE_NITF.equals(type.getValue())
+        && !(boolean) properties.getOrDefault(NITF_PROCESSING_KEY, false);
   }
 
-  private void process(List<ContentItem> contentItems) {
+  private void createContentItems(Metacard metacard) throws IOException {
+    ResourceResponse response;
+    try {
+      response = catalogFramework.getLocalResource(new ResourceRequestById(metacard.getId()));
+    } catch (ResourceNotFoundException | ResourceNotSupportedException | IOException e) {
+      LOGGER.warn("Error retrieving resource for thumbnail creation", e);
+      return;
+    }
+
+    ByteSource byteSource =
+        new ByteSource() {
+          @Override
+          public InputStream openStream() throws IOException {
+            return response.getResource().getInputStream();
+          }
+        };
+    ContentItem contentItem =
+        new ContentItemImpl(
+            metacard.getId(),
+            byteSource,
+            IMAGE_NITF,
+            null,
+            response.getResource().getInputStream().available(),
+            metacard);
     List<ContentItem> newContentItems = new ArrayList<>();
-    contentItems.forEach(contentItem -> process(contentItem, newContentItems));
-    contentItems.addAll(newContentItems);
+    process(contentItem, newContentItems);
+
+    Map<String, Serializable> properties = new HashMap<>();
+    properties.put(NITF_PROCESSING_KEY, true);
+    UpdateStorageRequest updateStorageRequest =
+        new UpdateStorageRequestImpl(newContentItems, properties);
+    try {
+      catalogFramework.update(updateStorageRequest);
+    } catch (IngestException | SourceUnavailableException e) {
+      LOGGER.warn("Error storing thumbnail/overview", e);
```

Comment:
`thumbnail/overview/original` or calculate message based on `createOverview` and `storeOriginalImage`.